### PR TITLE
Feature/exit codes

### PR DESCRIPTION
**Pull request by 5ebastianMeier**
This adds two options `exit-on-error` and `exit-on-aggregated-error`.

`exit-on-error` directly cancels everything with `process.exit(1)` as soon as an error occurs on a sub command.

`exit-on-aggregated-error` waits until all sub commands have run and then exits with `process.exit(1)`.

I added the options, because without them I didn't find any way to have the exit code on the original `meta` call reflect that errors occured on one of the commands.

This PR also involves PRs for `meta-exec`, `meta-loop` and `loop` itself.

I didn't clean up the commit history and I had to do some version changes during the process to test the changes on my local setup.

I'll gladly clean up everything as soon as you reviewed it, but would like to keep it until then for testing purposes. If you want I can also provide you with a test setup if that might be an issue (wasn't easy for me ;).

### CODE REVIEW

File: bin/meta [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env node
 
 const chalk = require('chalk');
-const debug = require('debug')('meta');
```
Comment (mateodelnorte) [inline bin/meta]: Can you nuke this change, please?

**Comment by mateodelnorte** [comment]
Mind adding yarn-error.log to .gitignore, while you're at it? :D

**Comment by 5ebastianMeier** [comment]
Will do ;)

File: bin/meta (hunk 1 above)
Comment (mateodelnorte) [reply bin/meta]: Ping ping on this change `const debug = require('debug')('@essential-projects/meta');`. Can you reset it to `meta`?

Comment (5ebastianMeier) [reply bin/meta]: Sorry totally missed this

**Comment by mateodelnorte** [comment]
fixed in 887636de05841690b9ad4f069b11a6cb1e44a155. only difference is renaming option to --exit-on-aggregated-error

**Pull request closed without merging**